boto/gs/key.py

```diff
@@ -707,7 +713,12 @@ def set_contents_from_string(self, s, headers=None, replace=True,
         self.md5 = None
         self.base64md5 = None
 
-        fp = StringIO(get_utf8_value(s))
+        if six.PY3:
```

Comment:
Looking at `get_utf8_value`, I don't think this does anything different in Python 3 than in 2.  I think we can revert this block of changes.